### PR TITLE
Add day(interval) Presto function

### PR DESCRIPTION
**Pull request by mbasmanova**
Differential Revision: D54830164

### CODE REVIEW

File: velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp [hunk 1]
```diff
@@ -804,6 +814,23 @@ TEST_F(DateTimeFunctionsTest, dayOfMonthDate) {
...
+    auto result2 = evaluateOnce<int64_t, int64_t>(
+        "day(c0)", {millis}, {INTERVAL_DAY_TIME()});
+
+    EXPECT_EQ(result, result2);
+    return result;
+  };
+
+  EXPECT_EQ(1, day(kMillisInDay));
+  EXPECT_EQ(1, day(kMillisInDay + kMillisInHour));
+  EXPECT_EQ(10, day(10 * kMillisInDay + 7 * kMillisInHour));
```
Comment (kagamiori) [inline velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp:831]: Can interval be negative? Shall we add some test cases with negative values?

**Review by kagamiori — APPROVED** [review]
LGTM. Thank you!

File: velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp:831]: It can. Added a test.

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks!

**Pull request closed without merging**